File: Java/Unit 02/Lec 14/Q2.java
Single File Programming Question
Problem Statement



Sophia is building a text-processing application that analyzes user-inputted sentences. 
  She needs to determine the number of words in each sentence entered by the user. 



Write a program for Sophia that takes a full sentence as input, splits it into 
  individual words, and outputs the total word count. 

Input format :
The input consists of a string, representing a sentence.

Output format :
The output prints an integer representing the number of words in the input sentence.



Refer to the sample output for formatting specifications.

Code constraints :
The given testcases fall under the following constraints:

The input sentence may include letters (both lowercase and uppercase), spaces, and common punctuation marks.

Sample test cases :
Input 1 :
The quick brown fox
Output 1 :
4
Input 2 :
Count the number of words in the program!
Output 2 :
8




  import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        // Read full sentence
        String sentence = sc.nextLine().trim();

        int wordCount = 0;

        if (!sentence.isEmpty()) {
            // Split sentence by one or more spaces
            String[] words = sentence.split("\\s+");
            wordCount = words.length;
        }

        // Print word count
        System.out.println(wordCount);

        sc.close();
    }
}

File: CPP /Unit 06/Lec 38/Q2.cpp
Single File Programming Question
Problem Statement



Suppose you are developing a program that takes input from users to determine the sentiment of their feedback. The program requires users to enter a numeric value representing their sentiment. The program uses exceptional handling to categorize the sentiment into three categories: positive, negative, or neutral.



Write a program to find whether the given number is Positive, Negative, or zero using the exceptional handling technique. Pass different parameters for different catch methods.



Note: This question helps in clearing TCS tests.

Input format :
The input consists of an integer.

Output format :
If the given integer is a positive integer, the output prints "You Entered Positive number".

If the given integer is a negative integer, the output prints "You Entered Negative number".

If the given integer is zero, the output prints "You Entered Zero".



Refer to the sample output for the formatting specifications.

Sample test cases :
Input 1 :
12
Output 1 :
You Entered Positive number
Input 2 :
-20
Output 2 :
You Entered Negative Number
Input 3 :
0
Output 3 :
You Entered Zero

#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    try {
        if (n > 0) {
            throw 1;
        } else if (n < 0) {
            throw -1;
        } else {
            throw 0;
        }
    }
    catch (int x) {
        if (x > 0) {
            cout << "You Entered Positive number";
        } else if (x < 0) {
            cout << "You Entered Negative Number";
        } else {
            cout << "You Entered Zero";
        }
    }

    return 0;
}
